src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -38,7 +39,76 @@ namespace app {
 namespace Clusters {
 namespace Thermostat {
 
-extern ThermostatAttrAccess gThermostatAttrAccess;
+CHIP_ERROR RemoveExpiredSuggestions(Delegate * delegate)
+{
+    VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    uint32_t currentMatterEpochTimestampInSeconds = 0;
+    CHIP_ERROR err                                = System::Clock::GetClock_MatterEpochS(currentMatterEpochTimestampInSeconds);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+    for (int i = static_cast<int>(delegate->GetNumberOfThermostatSuggestions() - 1);
+         i >= 0 && delegate->GetNumberOfThermostatSuggestions() > 0; i--)
+    {
+        ThermostatSuggestionStructWithOwnedMembers suggestion;
+
+        err = delegate->GetThermostatSuggestionAtIndex(i, suggestion);
+        if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
```

Comment:
If this happens it actually means the delegate somehow screwed up, right?

Reply:
I understand how GetThermostatSuggestionAtIndex works.

What I meant is that if GetThermostatSuggestionAtIndex returned CHIP_ERROR_PROVIDER_LIST_EXHAUSTED here, that means we are indexing past the end of the list as far as the delegate is concerned, right?  But we are starting at `GetNumberOfThermostatSuggestions() - 1` and going down from there, so how can we be past end of list?  Only if the delegate lied about `GetNumberOfThermostatSuggestions()` or removed more than one thing when we told it to remove or something.

As in, getting CHIP_ERROR_PROVIDER_LIST_EXHAUSTED here is NOT normal and we should probably not treat it as a "no error" condition?